storage/BloomFilterIndexSubBlock.hpp

```diff
@@ -0,0 +1,184 @@
+/**
+*   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+*     University of Wisconsin—Madison.
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+**/
+
+#ifndef QUICKSTEP_STORAGE_BLOOM_FILTER_INDEX_SUB_BLOCK_HPP_
+#define QUICKSTEP_STORAGE_BLOOM_FILTER_INDEX_SUB_BLOCK_HPP_
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "expressions/predicate/PredicateCost.hpp"
+#include "storage/IndexSubBlock.hpp"
+#include "storage/StorageBlockInfo.hpp"
+#include "storage/SubBlockTypeRegistryMacros.hpp"
+#include "utility/BloomFilter.hpp"
+#include "utility/Macros.hpp"
+
+namespace quickstep {
+
+class CatalogRelationSchema;
+class ComparisonPredicate;
+class IndexSubBlockDescription;
+class TupleIdSequence;
+class TupleStorageSubBlock;
+
+QUICKSTEP_DECLARE_SUB_BLOCK_TYPE_REGISTERED(BloomFilterIndexSubBlock);
+
+/** \addtogroup Storage
+*  @{
+*/
+
+/**
+* @brief An IndexSubBlock which implements a Bloom Filter index.
+* @note This IndexSubBlock supports both fixed-length attributes, and the
+*          variable length attributes.
+**/
+class BloomFilterIndexSubBlock : public IndexSubBlock {
+ public:
+  enum class BloomFilterSelectivity {
```

Comment:
Please add comments here describing what the enums mean. It would especially help with the All case since that was unclear until reading into the code.